resources/glossary/machine-learning.mdx

```diff
@@ -0,0 +1,45 @@
+
+## TYPES OF MACHINE LEARNING
+
+The three most important approaches in machine learning are supervised learning, unsupervised learning and reinforcement learning. Other methods such as [data mining](https://hash.ai/wiki/data-mining), semi-supervised learning or feature learning make use of those three essential methods listed above. Although Deep Learning has been the most popular paradigm in Machine Learning in the past 5 years, traditionally machine learning has made extensive use of decision trees, support-vector machines, bayesian networks and genetic algorithms. 
+
+1. **Supervised Learning**: the model learns from a training data with labels in order to be able to predict the labels of some test data, where the algorithm has not seen the labels. This can be for instance a classifier being able to tell if an image is a dog or a cat after having been exposed to millions of images of cats and dogs, with the corresponding labels. _Supervised_ comes from the model learning from the correct questions (training examples) and answers (training labels), like if it was a teacher supervising the learning by providing some material.
+
+2. **Unsupervised Learning**: the model does not have access to any label. One popular application is clustering where the goal is to identify natural groups of examples in the dataset. This could be for instance determining the behavior of a website user. Another idea is to see which datapoints in our dataset is very far away from what we would expect from the rest of the distribution, which we call an **outlier**, or **anomaly** in anomaly detection. This is used in computer security to detect strange behaviors, for instance with non human-like requests.
+
+3. **Reinforcement Learning**: an agent interacts with an environment by observing states and rewards from his actions. It shares some components of [agent-based modeling](https://hash.ai/wiki/agent-based-modeling) like having agents that interact with environments, and the state is essentially the collection of all the agents' properties. However, in reinforcement learning with one agent we do not consider the rules as something separate from the environment, but solely as a particular part of the environment that an agent needs to learn. In Multi-Agent Reinforcement Learning we can either consider each agent as being in its own one-agent reinforcement learning environment, or provide ways of communication between agents that would allow them to share their plans, states and rewards. For more details about reinforcement learning and how to apply it with neural networks see the glossary page on [deep reinforcement learning](https://hash.ai/wiki/deep-reinforcement-learning).
```

Comment:
```suggestion
3. **Reinforcement Learning**: an agent interacts with an environment by observing states and rewards from his actions. It shares some components of [agent-based modeling](https://hash.ai/glossary/agent-based-modeling) like having agents that interact with environments, and the state is essentially the collection of all the agents' properties. However, in reinforcement learning with one agent we do not consider the rules as something separate from the environment, but solely as a particular part of the environment that an agent needs to learn. In Multi-Agent Reinforcement Learning we can either consider each agent as being in its own one-agent reinforcement learning environment, or provide ways of communication between agents that would allow them to share their plans, states and rewards. For more details about reinforcement learning and how to apply it with neural networks see the glossary page on [deep reinforcement learning](https://hash.ai/glossary/deep-reinforcement-learning).
```